apps/notion-api/src/app/api/users/cron/run-sync-user/service.ts

```diff
@@ -0,0 +1,129 @@
+                { sync_started_at: 'asc' }, // Order by timestamp in ascending order
+            ],
+        });
+
+        if (!job) {
+            return;
+        }
+
+        const integration  = await prisma.integration.findFirst({
+            where: {
+                id: job.integration_id,
+            },
+        });
+
+        if (!integration) {
+            return;
+        }
+
+        const pageSize = process.env.USERS_SYNC_JOB_BATCH;
+        const notionUsersUrl = `https://api.notion.com/v1/users?page_size=${pageSize}${
+            job.pagination_token ? `&start_cursor=${job.pagination_token}` : ''
+        }`;
+
+        const response = await fetch(notionUsersUrl, {
+            method: 'GET',
+            headers: {
+                'Notion-Version': '2022-06-28',
+                Accept: 'application/json',
+                'Content-Type': 'application/json',
+                Authorization: `Bearer ${integration.notionToken}`,
+            },
+        });
+
+        const data : UsersResponseType = await response.json();
+        if ('status' in data) {
+            return;
+        }
+
+        const users: UserData[] = [];
+        for (const user of data.results) {
+            const userData: UserData = {
+                id: user.id,
+                displayName: user.name,
+                email: user.person.email,
+            }
+            users.push(userData);
+        }
+
+        const ELBA_API_BASE_URL = process.env.ELBA_API_BASE_URL;
+        const sourceId = process.env.ELBA_SOURCE_ID;
+        const optionsForUserUpdate = {
+            method: 'POST',
+            url: `${ELBA_API_BASE_URL}/api/rest/users`,
+            headers: {
+                accept: 'application/json',
+                'content-type': 'application/json',
+            },
+            data: { users, organisationId: integration.organization_id , sourceId},
+        };
+        await axios.request(optionsForUserUpdate);
+        
+        const nextCursor = data.next_cursor;
+
+        if (nextCursor) {
+            await prisma.users_Sync_Jobs.update({
+                where: {
+                    id: job.id,
+                },
+                data: {
+                    sync_started_at: new Date(),
+                    pagination_token: nextCursor ? nextCursor : '',
+                },
+            });
+        } else {
+            await prisma.users_Sync_Jobs.delete({
+                where: {
+                    id: job.id,
+                },
+            });
+
+            const options = {
+                method: 'DELETE',
+                url: `${ELBA_API_BASE_URL}/api/rest/users`,
+                headers: {
+                accept: 'application/json',
+                'content-type': 'application/json',
+                },
+                data: { users, organisationId: integration.organization_id , sourceId},
+            };
+            await axios.request(options);
+        }
+    } catch (error) {
+        
```

Comment:
Here is some pseudo code on how handle error in this case:
```suggestion
        if (error is NotionAPIError && error.retryAfter) {
          // update the job retry_after with the retrieved value
          return { message: 'Job delayed due to Notion rate limit', job };
        }
        // handle unauthorized access - the user might have disable the integration on notion
        if (error is NotionAPIError && error.response.status === 401) {
          // send a request to update the connection status on elba
          // delete the current job
          return { message: 'Job removed due to Notion unauthorized access', job }
        }
        // for the others (unexpected/unhandled) errors, we will increment the job retry count
        if (job.retry_count < process.env.MAX_USERS_SYNC_JOBS_RETRY_COUNT) {
          // increment job retry_count
          return { message: 'Job canceled due to retry limit', job }
        }
        // delete job
        throw error;
```

[Here is the documentation](https://elba-security.readme.io/reference/update-status) about /connection-status elba endpoint